File: trunk/ShadowFrameworkLite20/src/shadow/geometry/curves/SFTransformedCurve.java
package shadow.geometry.curves;

import shadow.geometry.SFCurve;
import shadow.math.SFTransform3f;
import shadow.math.SFValuenf;

public class SFTransformedCurve implements SFCurve{

	private SFCurve curve;
	private SFTransform3f trasnform;
	
	public SFTransformedCurve() {
		super();
	}

	public SFTransformedCurve(SFCurve curve, SFTransform3f trasnform) {
		super();
		this.curve = curve;
		this.trasnform = trasnform;
	}



	public void setCurve(SFCurve curve) {
		this.curve = curve;
	}



	public void setTrasnform(SFTransform3f trasnform) {
		this.trasnform = trasnform;
	}



	@Override
	public void destroy() {
		
	}  
	
	@Override
	public SFValuenf generateValue() {
		return curve.generateValue();
	}
	
	@Override
	public void get(float[] ts, SFValuenf[] vertices, SFValuenf[] derivatives) {
		curve.get(ts, vertices, derivatives);
		transfom(vertices, derivatives);
	}

	public void transfom(SFValuenf[] vertices, SFValuenf[] derivatives) {
		for (int i = 0; i < derivatives.length; i++) {
			trasnform.transformDir(derivatives[i]);
		}
		for (int i = 0; i < vertices.length; i++) {
			trasnform.transform(vertices[i]);
		}
	}
	
	@Override
	public void get(float[] ts, SFValuenf[] vertices, SFValuenf[] derivatives,
			SFValuenf[] derivatives2) {
		curve.get(ts, vertices, derivatives,derivatives2);
		for (int i = 0; i < derivatives2.length; i++) {
			trasnform.transformDir(derivatives2[i]);
		}
		transfom(vertices, derivatives);
	}
	
	
	@Override
	public void getDev2Dt(float ts, SFValuenf read) {
		curve.getDev2Dt(ts,read);
		trasnform.transformDir(read);
	}
	
	@Override
	public void getDev2Dt(float[] ts, SFValuenf[] read) {
		curve.getDev2Dt(ts, read);
		for (int i = 0; i < read.length; i++) {
			trasnform.transformDir(read[i]);
		}
	}
	
	@Override
	public void getDevDt(float t, SFValuenf read) {
		curve.getDevDt(t, read);
		trasnform.transformDir(read);
	}
	
	@Override
	public void getDevDt(float[] ts, SFValuenf[] read) {
		curve.getDevDt(ts, read);
		for (int i = 0; i < read.length; i++) {
			trasnform.transformDir(read[i]);
		}
	}
	
	@Override
	public float getTMax() {
		return curve.getTMax();
	}
	
	@Override
	public float getTMin() {
		return curve.getTMin();
	}
	
	@Override
	public void getVertex(float t, SFValuenf read) {
		curve.getVertex(t, read);
		trasnform.transform(read);
	}
	
	@Override
	public void getVertices(float[] ts, SFValuenf[] read) {
		curve.getVertices(ts, read);
		for (int i = 0; i < read.length; i++) {
			trasnform.transform(read[i]);
		}
	}
	
	@Override
	public void init() {
		// TODO Auto-generated method stub
		
	}
}

File: trunk/ShadowFrameworkJS/src/shadow/geometry/curves/SFTransformedCurve.js



function SFTransformedCurve(curve,transform){
	
}


SFTransformedCurve.prototype["setCurve"]=function(curve){
		this.curve = curve;
};

SFTransformedCurve.prototype["setTransform"]=function(curve){
		this.transfom = transfom;
};

SFTransformedCurve.prototype["destroy"]=function(){
		
};

SFTransformedCurve.prototype["generateValue"]=function(){
	return curve.generateValue();
};  
	
SFTransformedCurve.prototype["get"]=function(ts,vertices,derivatives){
	this.curve.get(ts, vertices, derivatives);
	this.transfomValues(vertices, derivatives);
};  
	
SFTransformedCurve.prototype["transfomValues"]=function(vertices,derivatives){
	for (var i = 0; i < derivatives.length; i++) {
		this.transfom.transformDir(derivatives[i]);
	}
	for (var i = 0; i < vertices.length; i++) {
		this.transfom.transform(vertices[i]);
	}
};  
	

SFTransformedCurve.prototype["get"]=function(ts,vertices,derivatives,derivatives2){
	this.curve.get2(ts, vertices, derivatives,derivatives2);
	for (var i = 0; i < derivatives2.length; i++) {
		this.trasnform.transformDir(derivatives2[i]);
	}
	this.transfomValues(vertices, derivatives);
};  
	

SFTransformedCurve.prototype["getDev2Dt"]=function(ts,read){
	this.curve.getDev2Dt(ts,read);
	this.trasnform.transformDir(read);
};  	
	

SFTransformedCurve.prototype["getDev2DtV"]=function(ts,read){
	this.curve.getDev2Dt(ts, read);
	for (var i = 0; i < read.length; i++) {
		this.transform.transformDir(read[i]);
	}
};  	
	
	
SFTransformedCurve.prototype["getDevDt"]=function(t,read){
	this.curve.getDevDt(ts,read);
	this.trasnform.transformDir(read);
};  

SFTransformedCurve.prototype["getDevDtV"]=function(ts,read){
	this.curve.getDevDt(ts, read);
	for (var i = 0; i < read.length; i++) {
		this.transform.transformDir(read[i]);
	}
};  		
	
SFTransformedCurve.prototype["getTMax"]=function(){
	return curve.getTMax();
};  
	
SFTransformedCurve.prototype["getTMin"]=function(){
	return curve.getTMin();
};  
	
SFTransformedCurve.prototype["getVertex"]=function(t,read){
	this.curve.getVertex(t, read);
	this.transform.transform(read);
};  

SFTransformedCurve.prototype["getVertices"]=function(ts,read){
	this.curve.getVertices(ts, read);
	for (var i = 0; i < read.length; i++) {
		this.transform.transform(read[i]);
	}
};  

SFTransformedCurve.prototype["init"]=function(){
	//nothing to do
};  
	
